### ISSUE TITLE
Same Certificate used for all generated profiles

### ISSUE THREAD

**Issue by Phoenix1o1** [Nyr/openvpn-install#678]
Ideally, you should generate a unique certificate for each client. 
It seems like this script somehow gives the same certificate to all profiles.
This causes issues because all the devices with the same certificate will be given the same internal IP and communication between different devices will not be possible.
A temporary fix for me has been the addition of the "duplicate-cn" line in the server.conf, but a more robust solution should be implemented.
(note that also client-to-client should be added to server.conf in order to enable cross communication)
BTW, thanks for your amazing job, this script has saved me in almost infinite occasions.

**Comment by Phoenix1o1** [comment]
I'm running 2 different VPS system, and i just checked on both of them. Even if i create 2 new profiles, and use say, my phone and my laptop to connect to my server, even from 2 different network, they both get assigned the same internal IP. Only by adding "duplicate-cn" I was able to get 2 different IP assigned. Why do you think this happens?

**Comment by Nyr** [comment]
The code from the script is very clear, a new certificate is generated for each client. Whatever is wrong, must be in your side.

You can try to generate two certificates, read them with cat and paste the entire, uncensored output here.

**Comment by Phoenix1o1** [comment]
Please find attached the 2 new certificates.

[Nas.txt](https://github.com/Nyr/openvpn-install/files/3832174/Nas.txt)
[P30Pro.txt](https://github.com/Nyr/openvpn-install/files/3832175/P30Pro.txt)


Maybe this comment from [this ](https://forums.openvpn.net/viewtopic.php?t=7486)discussion helps:

> 

Re: Two clients got the same IP
Post by max@OpenVPN » Fri Jan 07, 2011 6:59 pm

Hi Stephan,
thanks a lot for your support.
I've got the resolution:
For each client, I generated a unique certificate/key, but I always used the same CN. So, when the second client connected, the first client was dropped, because of identical CN OpenVPN thought, client 1 and 2 were one client.
Kind regards,
Markus
>

**Comment by Nyr** [comment]
Ask Angristan or use my installer, this issue tracker is for my project only and I don't support his fork.

**Comment by WaterSibilantFalling** [comment]
I don't know if this is relevant, but I get the errors (?) :


```
e is 65537 (0x010001)
Can't load /etc/openvpn/server/easy-rsa/pki/.rnd into RNG
139663193220224:error:2406F079:random number generator:RAND_load_file:Cannot open file:../crypto/rand/randfile.c:98:Filename=/etc/openvpn/server/easy-rsa/pki/.rnd

Using SSL: openssl OpenSSL 1.1.1d  10 Sep 2019
Generating a RSA private key
.................................................................................................................................................................................................................................+++++
.....+++++
writing new private key to '/etc/openvpn/server/easy-rsa/pki/private/server.key.1lCFzam2IY'

```

Is it possible that the random number generator can't be accessed, so identical certificates are produced?

**Comment by TinCanTech** [comment]
> Is it possible that the random number generator can't be accessed, so identical certificates are produced?

No, this is only a seeding mechanism for openssl.  However, openssl has protections against generating identical certificates, otherwise VPNs would be kind of dead ..

Easyrsa, are considering what is the best solution to fix this warning but it is generally not worth worrying about.
